src/mdns/mdns_avahi.cpp

```diff
@@ -843,9 +767,38 @@ otbrError PublisherAvahi::TxtListToAvahiStringList(const TxtList &   aTxtList,
     return error;
 }
 
-std::string PublisherAvahi::MakeFullName(const std::string &aName)
+Publisher::ServiceRegistrationPtr PublisherAvahi::FindServiceRegistration(const AvahiEntryGroup *aEntryGroup) const
 {
-    return aName + "." + kDomain;
+    ServiceRegistrationPtr result = nullptr;
+
+    for (auto &kv : mServiceRegistrations)
+    {
+        auto &serviceReg = static_cast<AvahiServiceRegistration &>(*kv.second);
```

Comment:
```suggestion
    for (const auto &kv : mServiceRegistrations)
    {
        const auto &serviceReg = static_cast<const AvahiServiceRegistration &>(*kv.second);
```

nit. Feel free to leave it as is.